<launch>

	<arg name="map_file" default="$(find agv_control)/maps/2014-07-03-12-33-ei-planning.yaml"/>
	<!--<arg name="map_file" default="$(find agv_control)/maps/gallery_NEW_corridor.yaml"/>-->
	
	<arg name="rviz_file" default="$(find agv_control)/RViz/rviz_gallery_sim.vcg" />
	

	<arg name="mat_file" default="$(find agv_control)/params/params_lgv1000.mat"/>

	
	<arg name="missions_file" default="$(find agv_control)/missions/missions.txt"/>
	
	
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" >
    	<param name="frame_id" value="/map" />
    </node>

	<!-- Vehicle names and priorities -->
	<rosparam param="vehicleNames">['Alfa','Bravo', 'Charlie', 'Delta', 'Echo', 'Foxtrot']</rosparam>
	<rosparam param="vehiclePriorities">[1, 2, 3, 4, 5, 6]</rosparam>
	
	<!-- Pure pursuit parameters for simulation-->
	<rosparam param="lookAheadDistance">2</rosparam>
	<rosparam param="linVelocity">0.2</rosparam>
	<rosparam param="maxAngVelocity">0.4</rosparam>
	
	<!-- Pure pursuit parameters for real Pioneers-->
	<!--<rosparam param="lookAheadDistance">0.2</rosparam>
	<rosparam param="linVelocity">0.1</rosparam>
	<rosparam param="maxAngVelocity">0.1</rosparam>-->
	
	<rosparam param="posTolerance">0.7</rosparam>

        <node pkg="tf" type="static_transform_publisher" name="laser_tf2" args="0.529 0 0 0 0 0 /Foxtrot/base_link /Foxtrot/laser 50"/>

    <node pkg="agv_control" name="pathPlanner" type="pathPlanner" args="$(arg mat_file)" />
    
 
    <node pkg="agv_control" name="agvController" type="agvController" ns="Foxtrot" args= "$(arg mat_file)" output="screen" >
    	<remap from="/Foxtrot/map" to="/map"/>
    	<remap from="/Foxtrot/getPlanSrv" to="/getPlanSrv"/>
    </node>
    

    
    <!--<include file="$(find agv-control)/launch/pioneer3dx_urdf_AGV.launch">
    	<arg name="robot_name" value="Foxtrot"/>
    </include>-->
    
   
    
    <node pkg="agv_control" name="motionSimSR" type="motionSimSR" output="screen"/>
    
    
    <!--<node pkg="agv_control" name="missionPublisher" type="missionPublisher" args= "-mf $(arg missions_file)"/> -->
    
</launch>
